packages/frontend/src/Views/Home/Pagination/PageSetter/PageSetter.tsx

```diff
@@ -0,0 +1,85 @@
+import { Wrapper, ArrowIcon, PageNumber } from "./PageSetter.styled";
+import paginationarrow from "src/assets/images/paginationarrow.svg";
+import { setPage } from "../reducer/actions/pagination.actions";
+import { usePaginationContext } from "../hooks/usePaginationContext";
+import { Color } from "src/enums/color.enum";
+import { GetVisisblePagesArrayParams } from "./PageSetter.types";
+
+export const PageSetter: React.FC = () => {
+  const [{ maxPages, page }, dispatch] = usePaginationContext();
+
+  const getVisiblePagesArray = ({ activePage, totalPages }: GetVisisblePagesArrayParams) => {
```

Comment:
na zewnątrz jako pure function, jeśli nie potrzebujesz definiować funkcji w komponencie to tego nie rób po pierwsze zwiększysz czytelnośc po drugie, przy każdym kolejnym renderze do contsa get visible arra przypisywana jest nowa referencja o ile przy jendym komponencie to nie jest probelm to jakbyś miał 100 paginatorów to mogłoby juz to mieć znaczenie